valhalla/proc/costing_params.py

```diff
@@ -239,15 +239,14 @@ def set_costing_options(self, proc_algo, parameters, context):
                 )
             )
 
-        # Boolean hazmat
         setattr(
             self,
             TRUCK_COSTING.HAZMAT,
-            proc_algo.parameterAsBool(
+            bool(proc_algo.parameterAsBool(
```

Comment:
it's fine to cast again, but `parameterAsBool` should already return a `bool`: https://qgis.org/pyqgis/3.0/core/Processing/QgsProcessingAlgorithm.html#qgis.core.QgsProcessingAlgorithm.parameterAsBool. if not, there's smth wrong with pyqgis.